companies/permissions.py

```diff
@@ -0,0 +1,18 @@
+from rest_framework.permissions import BasePermission
+from authentication.models import CompanyAndUserRelation
+
+class EditCopmanyPermission(BasePermission):
+    """
+    Custom permission to allow editing company objects only for users who have a relation to the company.
+    """
+        
+    def has_object_permission(self, request, view, obj):
+        """
+        Return `True` if the requesting user has a relation to the company (obj), `False` otherwise.
+        """  
+        try:
+            CompanyAndUserRelation.objects.get(user_id=request.user.user_id, company_id=obj.company_id)
```

Comment:
user=request.user - this is not working and I'm getting "Cannot resolved keyword 'user' into field" error. Also could you please explain what  does it mean “leverages the ORM's capabilities”?

Reply:
In this context, "leverages the ORM's capabilities" refers to the more efficient and idiomatic use of Django's ORM system when dealing with relationships between models, specifically between CompanyAndUserRelation and User.
When you have a `ForeignKey` in your model that points to another model (in this case, `CompanyAndUserRelation` has a `ForeignKey` to `User`), Django's ORM allows you to work directly with model instances rather than foreign key IDs. This approach has several benefits (readability, simplicity, integrity etc.)
_Example:_
```
# Less idiomatic, manually dealing with IDs
company_user_relation = CompanyAndUserRelation(user_id=request.user.user_id)
company_user_relation.save()

# More idiomatic, leveraging Django's ORM capabilities
company_user_relation = CompanyAndUserRelation(user=request.user)
company_user_relation.save()
```